jena-arq/src/main/java/org/apache/jena/sparql/engine/main/VarFinder.java

```diff
@@ -329,12 +336,43 @@ public void visit(OpExtend opExtend) {
 
         private void processAssignVarExprList(VarExprList varExprList) {
             varExprList.forEachVarExpr((v,e)-> {
-                defines.add(v) ; // Expression may eval to error -> unset?
-                if ( e != null )
+                if ( e != null ) {
+                    boolean isLikelyToBeDefined = isLikelyToBeDefined(e);
+                    if (isLikelyToBeDefined) {
+                        defines.add(v) ; // Expression may eval to error -> unset?
```

Comment:
If the variable may be unbound it probably still needs to be tracked - e.g. in optDefines?

Reply:
Yes - that seems likely. 
Not putting it in is quite safe.
Doing so and tests all pass.

Reply:
Done